sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReceiveLinkProcessor.java

```diff
@@ -199,12 +198,7 @@ public void onNext(ServiceBusReceiveLink next) {
             oldSubscription = currentLinkSubscriptions;
 
             currentLink = next;
-            next.setEmptyCreditListener(() -> {
-                final int creditsToAdd = getCreditsToAdd(0);
-                linkCreditsAdded.set(creditsToAdd > 0);
-
-                return creditsToAdd;
-            });
+            next.setEmptyCreditListener(() -> 0);
```

Comment:
Should we do similar logic in SessionManager, Like we spoke over the chat.

Reply:
Updated to 0 in ServiceBusSessionReceiver